pkg/cmd/admin/policy/psp_review.go

```diff
@@ -0,0 +1,302 @@
+	Namespace               string
+	Filename                string
+
+	PodTemplateSpec *kapi.PodTemplateSpec
+	User            string
+	Groups          []string
+	ServiceAccount  string
+
+	PrintObject func(obj runtime.Object, out io.Writer) error
+}
+
+func (o *subjectReviewOptions) Run(out io.Writer) error {
+	var (
+		response runtime.Object
+		err      error
+	)
+	userOrSA := o.User
+	if len(o.ServiceAccount) > 0 {
+		userOrSA = o.ServiceAccount
+	}
+	switch {
+	case len(userOrSA) > 0 || len(o.Groups) > 0:
+		podSecurityPolicySubjectReview := &securityapi.PodSecurityPolicySubjectReview{
+			Spec: securityapi.PodSecurityPolicySubjectReviewSpec{
+				Template: *o.PodTemplateSpec,
+				User:     userOrSA,
+				Groups:   o.Groups,
+			},
+		}
+		response, err = o.subjectReviewClient.PodSecurityPolicySubjectReviews(o.Namespace).Create(podSecurityPolicySubjectReview)
+		if err != nil {
+			return err
+		}
+	default:
+		podSecurityPolicySelfSubjectReview := &securityapi.PodSecurityPolicySelfSubjectReview{
+			Spec: securityapi.PodSecurityPolicySelfSubjectReviewSpec{
+				Template: *o.PodTemplateSpec,
+			},
+		}
+		response, err = o.selfSubjectReviewClient.PodSecurityPolicySelfSubjectReviews(o.Namespace).Create(podSecurityPolicySelfSubjectReview)
+		if err != nil {
+			return err
+		}
+	}
+	return o.PrintObject(response, out)
+}
+
+func (o *subjectReviewOptions) Complete(f *clientcmd.Factory, args []string, cmd *cobra.Command) error {
+	if len(o.User) > 0 && len(o.ServiceAccount) > 0 {
+		return fmt.Errorf("--user and --serviceaccount are mutually exclusive")
+	}
+	namespace, enforceNamespace, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+	o.Namespace = namespace
+	oclient, _, err := f.Clients()
+	if err != nil {
+		return fmt.Errorf("unable to obtain client: %v", err)
+	}
+	o.subjectReviewClient = oclient
+	o.selfSubjectReviewClient = oclient
+
+	o.Namespace = namespace
+	mapper, typer := f.Object()
+	infos, err := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+		NamespaceParam(namespace).
+		FilenameParam(enforceNamespace, &resource.FilenameOptions{Recursive: false, Filenames: []string{o.Filename}}).
+		SingleResourceType().
+		Do().
+		Infos()
+
+	if err != nil {
+		return fmt.Errorf("unable to find a resource: %v", err)
+	}
+	if len(infos) != 1 {
+		return fmt.Errorf("expected a resource for path: %s", o.Filename)
+	}
+	o.PodTemplateSpec, err = returnPodTemplateSpec(infos[0].Object)
+	if err != nil {
+		return fmt.Errorf("unable to find podTemplateSpec from %s: %v", o.Filename, err)
+	}
+
+	o.PrintObject = func(c *cobra.Command, m meta.RESTMapper) func(runtime.Object, io.Writer) error {
+		return func(obj runtime.Object, out io.Writer) error {
+			return f.PrintObject(c, m, obj, out)
+		}
+	}(cmd, mapper)
+
+	return nil
+}
+
+func NewCmdSubjectReview(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
```

Comment:
\<me ranting>
It's nice and readable to have the NewCmd* function right after *Options structure. 
\</me ranting>